### DISCUSSION TITLE
Frontend Initiatives Meeting - May 25, 2021

### DISCUSSION THREAD

**Discussion by varl** [dhis2/notes#286] [Meetings] [answered]
When: May 25, 2021 @ 15:00 CET \
Where: [Remote](https://calendar.google.com/event?action=TEMPLATE&tmeid=XzYwcjRhY2hoOGdvazZiOW44OTM0OGI5azY5MWowYmExNzUzNDRiOXA2Y3A0YWdxMjYwc2o0aGk2OGdfMjAyMTA1MjVUMTMwMDAwWiB2aWt0b3JAZGhpczIub3Jn&tmsrc=viktor%40dhis2.org)

# Agenda

-   General updates
    -   [GitHub changed the Personal Access Token formats](https://github.blog/2021-04-05-behind-githubs-new-authentication-token-formats/) so they need to be regenerated in the new format. If there are any problems related to `dhis2-bot` not having access, let us know.
    -   [Recommend backing off the amount of open PRs Dependabot](https://github.com/dhis2/cli-style/pull/404) maintains in one update cycle. They are using a lot of resources right now when they rebase and all workflows run again.
    -   We are looking at visual regression testing solutions for e.g. UI. If anyone has practical experience with e.g. [Percy](https://percy.io), [Chromatic](https://www.chromatic.com/), or others, let us know and we'd love to pick your brain a bit !
    -   `DataTable` PR status
    -   Browserlist support needs to be defined.
    -   Rolling meeting notes for UI meetings?

-   Working group updates
    -  PWA
    - Client-side caching
    - Cypress
    - PR deploy previews
    - CI/CD
    - Developer outreach

-   Others?

**Comment by ghost** [comment]
In a PR that updated cli-style I noticed that opera mini was causing a ton of warnings as it's a very bare-bones browser (it doesn't support promises for example). Temporary solution was to remove it from the browserslist.

However, I suspect that that'll also mean that babel-env won't consider it for polyfilling (as that uses browserslist, I'm not sure if it uses our root browserslist at the moment). If that's correct, that means that our apps are possibly not usable on opera mini (no polyfills means no promises and thus everything breaking).

Looking at the browser share for opera on mobile in Africa (https://gs.statcounter.com/browser-market-share/mobile/africa/#monthly-202004-202104) opera is the third largest browser (about 10%). Then again, maybe that's not representative of our users as our apps aren't really geared towards mobile usage anyways.

Would be good to talk about this and decide if we either want to support opera mini and polyfill (I can silence the lint warnings by adding the polyfills to the eslint settings), or if we can definitively drop it from the browserslist.

**Comment by edoardo** [comment]
`DataTable` PR status... 🙊

**Comment by ghost** [comment]
> We are looking at visual regression testing solutions for e.g. UI. If anyone has practical experience with e.g. Checkly, Chromatic, or others, let us know and we'd love to pick your brain a bit !

I think we can drop Checkly and add https://percy.io

**Comment by varl** [comment, marked as answer]
Meeting minutes PR: https://github.com/dhis2/notes/pull/287
